macros/benches/match.rs

```diff
@@ -0,0 +1,141 @@
+use criterion::black_box;
+use criterion::criterion_group;
+use criterion::criterion_main;
+use criterion::Criterion;
+
+use tinystr::{TinyStr16, TinyStr4, TinyStr8};
+use tinystr_macros::{tinystr16, tinystr4, tinystr8};
+
+macro_rules! count {
+    () => (0usize);
+    ( $x:tt $($xs:tt)* ) => (1usize + count!($($xs)*));
+}
+
+macro_rules! gen_match {
+    ($TinyType:ident, [$( ($item:ident, $tiny:expr) ),*]) => {
```

Comment:
@Manishearth @zbraniecki @filmil Is there a `macro_rules!` definition here where I can call the sub-macro like `tinystr4!` instead of repeating the strings, once as an identifier and once as a TinyStr?  In other words, at the call site, instead of

```rust
    gen_match!(TinyStr4, [
        (US, tinystr4!("US")),
        (GB, tinystr4!("GB")),
        (AR, tinystr4!("AR"))
    ]);
```

how can I make it

```rust
    gen_match!(TinyStr4, tinystr4!, [
        US,
        GB,
        AR
    ]);
```